src/core/dataset_container.jl

```diff
@@ -0,0 +1,179 @@
+struct DatasetContainer{T}
+    duals::Dict{ConstraintKey, T}
+    aux_variables::Dict{AuxVarKey, T}
+    variables::Dict{VariableKey, T}
+    parameters::Dict{ParameterKey, T}
+    expressions::Dict{ExpressionKey, T}
+end
+
+function DatasetContainer{T}() where {T <: AbstractDataset}
+    return DatasetContainer(
+        Dict{ConstraintKey, T}(),
+        Dict{AuxVarKey, T}(),
+        Dict{VariableKey, T}(),
+        Dict{ParameterKey, T}(),
+        Dict{ExpressionKey, T}(),
+    )
+end
+
+function Base.empty!(container::DatasetContainer)
+    for field in fieldnames(DatasetContainer)
+        field_dict = getfield(container, field)
+        for val in values(field_dict)
+            empty!(val)
+        end
+    end
+    @debug "Emptied the store" _group = LOG_GROUP_SIMULATION_STORE
+    return
+end
+
+function get_duals_values(container::DatasetContainer)
+    return container.duals
```

Comment:
These functions are going to return different types based on the underlying type. Is that ok?

Reply:
We can probably remove these methods in favor of others.

Reply:
I would need to refactor `_get_emulation_store_requirements` to be able to remove these methods.

Reply:
I ran a @code_warntype and it seems not to matter. I think we can leave it like that since this is not in the critical path

Reply:
I recommend defining this only for the dataframe version - `DatasetContainer{DataFrameDataset}`. We don't ever want to expose the values attribute of `HDF5Dataset` because of the caching layer in `Hdf5SimulationStore`.